frontend/app/utils/route-utils.ts

```diff
@@ -0,0 +1,10 @@
+import { useMatches } from '@remix-run/react';
+
+export function usePageTitle() {
+  return useMatches()
+    .map((match) => match.data)
+    .filter((data): data is { pageTitle?: string } => data !== undefined && data !== null)
+    .filter((data) => data?.pageTitle !== undefined && data?.pageTitle !== null)
+    .map((data) => data.pageTitle)
+    .reduce((last, curr) => curr ?? last, undefined);
+}
```

Comment:
Use `Zod` for the object validation and infer PageTitle object type with it.

```suggestion
export function usePageTitle() {
  return useMatches()
    .map(({ data }) => data as any)
    .map(({ data }) => {
      const result = pageTitle.safeParse(data);
      if (result.success) return result.data;
      return undefined;
    })
    .filter((item): item is PageTitle => item !== undefined)
    .map(({pageTitle}) => pageTitle)
    .reduce((last, curr) => curr ?? last, undefined); 
}
```

Reply:
Will do, friend.. hang tight.

Reply:
This guy.. who does he think he is?

Reply:
Oops wrong window.

Reply:
[Fixed](https://github.com/DTS-STN/canadian-dental-care-plan/pull/65/files#diff-fa420e73484360a54e4949f50d2e0c50929565e217a8d566ad69a266c692abc0R5-R12)